crates/bevy_ecs/src/system/system_piping.rs

```diff
@@ -406,6 +406,34 @@ pub mod adapter {
     /// ```
     pub fn ignore<T>(In(_): In<T>) {}
 
+    /// System adapter that inverts the output of the previous system in a pipe.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// // Building a new schedule/app...
+    /// let mut sched = Schedule::default();
+    /// sched.add_system(
+    ///         // This system will always run.
+    ///         my_system.run_if(always_false.pipe(system_adapter::not))
+    ///     )
+    ///     // ...
+    /// #   ;
+    /// # let mut world = World::new();
+    /// # sched.run(&mut world);
+    ///
+    /// // A system that always returns false.
+    /// fn always_false() -> bool {
+    ///    false
+    /// }
+    ///
+    /// # fn my_system() {}
+    /// ```
+    pub fn not(In(val): In<bool>) -> bool {
```

Comment:
Can you make this use `std::ops::Not`?

Reply:
I tried, but it will look something like this because I can't implement `Not` for `Into<T>, I need an additional trait to convert into condition:
```rust
.run_if(!my_condition.into_condition())
```

Reply:
Sorry I should have been more clear. Can you make this adapter adapter work with any `T: Not` type?

`pub fn not<T: std::ops::Not>(In(val): In<T>) -> T::Output`

Reply:
Make sense, done!